### PR TITLE
Revert "Revert "Support deployable instance"" #2880

### PR DESCRIPTION
**Pull request by droot**
This PR adds a fix for the broken test to the  [PR](https://github.com/GoogleContainerTools/kpt/pull/2880https://github.com/GoogleContainerTools/kpt/pull/2880)

### CODE REVIEW

**Comment by droot** [comment]
@martinmaly @yuwenma This is the PR that reverts the original PR (for supporting deployable instance). This also fixes another flaky test that was causing original revert PR failure (https://github.com/GoogleContainerTools/kpt/pull/2880)

**Comment by droot** [comment]
I have also created separate PR https://github.com/GoogleContainerTools/kpt/pull/2882 for the individual test (if you prefer that).

**Review by martinmaly — APPROVED** [review]
Thank you!

File: internal/builtins/testData/pkg-with-existing-ctx/in.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        internal.config.kubernetes.io/seqindent: 'compact'
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: kptfile.kpt.dev
+      annotations:
+        config.kubernetes.io/local-config: "true"
+        internal.config.kubernetes.io/path: 'package-context.yaml'
+    data:
+      name: order-service
```
Comment (yuwenma) [inline internal/builtins/testData/pkg-with-existing-ctx/in.yaml:28]: Is this part the generated biultin pkg-context? which means they should not in the `in.yaml` but only in the `out.yaml`?

**Review by yuwenma — APPROVED** [review]

File: internal/builtins/testData/pkg-with-existing-ctx/in.yaml (hunk 1 above)
Comment (droot) [reply internal/builtins/testData/pkg-with-existing-ctx/in.yaml:28]: This is testing idempotency of the `gen-pkg-context` function which ensures correct pkg context is generated even if input contains pkg-context.yaml.

File: internal/cmdget/cmdget.go [hunk 2]
```diff
@@ -54,6 +54,8 @@ func NewRunner(ctx context.Context, parent string) *Runner {
 	c.Flags().StringVar(&r.strategy, "strategy", string(kptfilev1.ResourceMerge),
 		"update strategy that should be used when updating this package -- must be one of: "+
 			strings.Join(kptfilev1.UpdateStrategiesAsStrings(), ","))
+	c.Flags().BoolVar(&r.isDeploymentInstance, "for-deployment", false,
+		"(Experimental) indicates if this package will be deployed to a cluster.")
```
[the remark this answers is not in the document]
Comment (droot) [reply internal/cmdget/cmdget.go:58]: Yes, can be added later.

**Pull request merged**